pkg/cmd/fmt.go

```diff
@@ -0,0 +1,68 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/snyk/driftctl/pkg/analyser"
+	"github.com/spf13/cobra"
+
+	"github.com/snyk/driftctl/pkg"
+	"github.com/snyk/driftctl/pkg/cmd/scan/output"
+)
+
+func NewFmtCmd(opts *pkg.FmtOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:    "fmt",
+		Long:   "Take an analysis results in JSON on stdin and return it in another format",
+		Hidden: true,
+		Args:   cobra.NoArgs,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			outputFlag, _ := cmd.Flags().GetStringSlice("output")
+			if len(outputFlag) > 1 {
+				return errors.New("Only one output format can be set")
+			}
+			out, err := parseOutputFlags(outputFlag)
+			if err != nil {
+				return err
+			}
+			opts.Output = out[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return runFmt(opts, os.Stdin)
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringSliceP(
+		"output",
+		"o",
+		[]string{output.Example(output.ConsoleOutputType)},
+		"Output format, by default it will write to the console\n"+
+			"Accepted formats are: "+strings.Join(output.SupportedOutputsExample(), ",")+"\n",
+	)
+
+	return cmd
+}
+
+func runFmt(opts *pkg.FmtOptions, reader io.Reader) error {
+
+	var analysisText []byte
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		analysisText = append(analysisText, scanner.Bytes()...)
+	}
+
+	analysis := analyser.NewAnalysis(analyser.AnalyzerOptions{})
```

Comment:
We don't use the Deep option here, this is going to cause the `--deep` flag being ignored, isn't it ?

Reply:
We are only un-marshalling stdin to the struct here.
Whether if we set any option it will not change anything since we are not  running the analyzer, options will never be used.

Reply:
Right, but for example we use the deep option in the console output to alter the result. This will not work with the fmt command since we never use it. For now the impact is very low, but someday we could have inconsistency between the scan and fmt command since it lacks of all the options. Dunno if it makes sense?

Reply:
Hmm good point, we should then serialize also analysis options to be consistent.

I need to add that serialization in that PR, thanks for raising that 🙏🏻